### PR TITLE
[OHIF-288] Rehydrating SRs is missing Labels

### PR DESCRIPTION
**Pull request by JamesAPetts**
- Encode measurements  as `Finding Site`s and and additional findings as `Finding`s.
- Parse these and show these labels in the SR viewport.
- Consume these labels upon rehydration to achieve the same state as prior to saving the SR.
- SRs made prior to this PR still work.

Sister PR: https://github.com/dcmjs-org/dcmjs/pull/135

### CODE REVIEW

File: extensions/dicom-sr/src/OHIFCornerstoneSRViewport.js [hunk 1]
```diff
@@ -374,8 +375,12 @@ function OHIFCornerstoneSRViewport({
           imageId
         );
 
+        debugger;
```
Comment (dannyrb) [inline extensions/dicom-sr/src/OHIFCornerstoneSRViewport.js:378]:
```suggestion
```

File: extensions/dicom-sr/src/OHIFCornerstoneSRViewport.js [hunk 2]
```diff
@@ -374,8 +375,12 @@ function OHIFCornerstoneSRViewport({
...
+
```
Comment (dannyrb) [inline extensions/dicom-sr/src/OHIFCornerstoneSRViewport.js:379]:
```suggestion
```

File: extensions/dicom-sr/src/getSopClassHandlerModule.js [hunk 3]
```diff
@@ -25,13 +25,19 @@ const CodeNameCodeSequenceValues = {
   TrackingUniqueIdentifier: '112040',
   TrackingIdentifier: '112039',
   Finding: '121071',
+  FindingSite: 'G-C0E3', // SRT
+  CornerstoneFreeText: 'CORNERSTONEFREETEXT', // CST4
+};
+
+const CodingSchemeDesignators = {
+  SRT: 'SRT',
```
Comment (dannyrb) [inline extensions/dicom-sr/src/getSopClassHandlerModule.js:33]: What is "SRT"?

File: extensions/dicom-sr/src/tools/DICOMSRDisplayTool.js [hunk 4]
```diff
@@ -280,10 +280,9 @@ export default class DICOMSRDisplayTool extends BaseTool {
 }
 
 function _getTextBoxLinesFromLabels(labels) {
-  // TODO -> max 2 for now, need a generic solution for this!
-
-  const labelLength = Math.min(labels.length, 2);
+  // TODO -> max 3 for now (label + shortAxis + longAxis), need a generic solution for this!
```
Comment (dannyrb) [inline extensions/dicom-sr/src/tools/DICOMSRDisplayTool.js:283]: Do we have a follow-up issue for this? If so, can we add it in comments? If not, can we create it?

File: extensions/dicom-sr/src/utils/getLabelFromDCMJSImportedToolData.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+export default function getLabelFromDCMJSImportedToolData(toolData) {
```
Comment (dannyrb) [inline extensions/dicom-sr/src/utils/getLabelFromDCMJSImportedToolData.js:1]: JSDoc explaining what this is, what it does, why we do it?

File: extensions/dicom-sr/src/utils/getToolStateToCornerstoneMeasurementSchema.js [hunk 6]
```diff
@@ -119,7 +121,11 @@ function EllipticalRoi(measurementData, imageId, _getValueTypeFromToolType) {
     StudyInstanceUID,
   } = instance;
 
-  const { start, end } = measurementData.handles;
+  const { handles, label } = measurementData;
+
+  debugger;
```
Comment (dannyrb) [inline extensions/dicom-sr/src/utils/getToolStateToCornerstoneMeasurementSchema.js:126]:
```suggestion
```

File: extensions/dicom-sr/src/utils/getToolStateToCornerstoneMeasurementSchema.js [hunk 7]
```diff
@@ -119,7 +121,11 @@ function EllipticalRoi(measurementData, imageId, _getValueTypeFromToolType) {
...
+
```
Comment (dannyrb) [inline extensions/dicom-sr/src/utils/getToolStateToCornerstoneMeasurementSchema.js:127]:
```suggestion
```

File: extensions/dicom-sr/src/utils/getToolStateToCornerstoneMeasurementSchema.js [hunk 8]
```diff
@@ -119,7 +121,11 @@ function EllipticalRoi(measurementData, imageId, _getValueTypeFromToolType) {
     StudyInstanceUID,
   } = instance;
 
-  const { start, end } = measurementData.handles;
+  const { handles, label } = measurementData;
+
```
Comment (dannyrb) [inline extensions/dicom-sr/src/utils/getToolStateToCornerstoneMeasurementSchema.js:125]:
```suggestion
```

File: extensions/measurement-tracking/src/_shared/createReportAsync.js [hunk 9]
```diff
@@ -18,7 +18,8 @@ async function createReportAsync(servicesManager, dataSource, measurements) {
   try {
     const naturalizedReport = await DICOMSR.storeMeasurements(
       measurements,
-      dataSource
+      dataSource,
+      ['ArrowAnnotate']
```
Comment (dannyrb) [inline extensions/measurement-tracking/src/_shared/createReportAsync.js:22]: What meow?

File: extensions/measurement-tracking/src/panels/PanelMeasurementTableTracking/index.js [hunk 10]
```diff
@@ -148,17 +148,14 @@ function PanelMeasurementTableTracking({ servicesManager, extensionManager }) {
...
     const measurements = MeasurementService.getMeasurements();
     const trackedMeasurements = measurements.filter(
       m =>
         trackedStudy === m.referenceStudyUID &&
         trackedSeries.includes(m.referenceSeriesUID)
     );
 
-    // TODO -> local download.
-    DICOMSR.downloadReport(trackedMeasurements, dataSource);
+    DICOMSR.downloadReport(trackedMeasurements, ['ArrowAnnotate']);
```
Comment (dannyrb) [inline extensions/measurement-tracking/src/panels/PanelMeasurementTableTracking/index.js:158]: Can we make this more obvious?

`const additionalFindings = ['ArrowAnnotate'];`

and then pass that in, so it's a named param.

File: extensions/dicom-sr/src/getSopClassHandlerModule.js (hunk 3 above)
Comment (JamesAPetts) [reply extensions/dicom-sr/src/getSopClassHandlerModule.js:33]: Its a DICOM coding scheme designator, its in the standard.

File: extensions/dicom-sr/src/tools/DICOMSRDisplayTool.js (hunk 4 above)
Comment (JamesAPetts) [reply extensions/dicom-sr/src/tools/DICOMSRDisplayTool.js:283]: I think its out of scope for the foresable future, but I can create the issue.

File: extensions/measurement-tracking/src/_shared/createReportAsync.js (hunk 9 above)
Comment (JamesAPetts) [reply extensions/measurement-tracking/src/_shared/createReportAsync.js:22]: We need to encode additional findings as findings, and measurements as findingSites in the DICOM, there isn't a good way of doing this extensibly without passing in the list of what should be an annotation, happy for any suggestions.

File: extensions/measurement-tracking/src/panels/PanelMeasurementTableTracking/index.js (hunk 10 above)
Comment (JamesAPetts) [reply extensions/measurement-tracking/src/panels/PanelMeasurementTableTracking/index.js:158]: gotcha.

**Pull request merged**